packages/@lightningjs/ui-components/src/components/Icon/Icon.js

```diff
@@ -64,25 +64,10 @@ export default class Icon extends Base {
       this.texture = null;
       return;
     }
-    const { icon, w, h } = this;
-    if (
-      !this.prevTemplateParams ||
-      !stringifyCompare({ icon, w, h }, this.prevTemplateParams)
-    ) {
-      this.prevTemplateParams = { icon, w, h };
-      const template = getIconTemplate(icon, w, h);
-      this.patch(template);
-      // only update color if color style is defined in theme
-      if (!template.texture && this.style.color) {
-        this.smooth = {
-          color: getValidColor(this.style.color)
-        };
-      }
-    } else if (this.style.color) {
-      this.color = getValidColor(this.style.color);
-    }
 
-    // setting the radius on the Icon component
+    const template = this._getIconTemplate();
+    this.patch(template);
```

Comment:
if the templates variable isn't be used anywhere else, maybe put it in one line?